modules/gdscript/language_server/gdscript_extend_parser.cpp

```diff
@@ -299,7 +299,7 @@ void ExtendGDScriptParser::parse_function_symbol(const GDScriptParser::FunctionN
 	r_symbol.range.start.line = line;
 	r_symbol.range.start.character = p_func->column;
 	r_symbol.range.end.line = MAX(p_func->body->end_line - 2, p_func->body->line);
```

Comment:
There is a bug on the GDScript parse with function body position here.
CC @vnen 
We'd better ensure the value is valid here maybe `CLAMP(p_func->body->end_line - 2, p_func->body->line, lines.size()-1)`

Reply:
CLAMP does not work, because when end_line is -1 the body line will be exact the lines.size().
That one worked with my testing setup: (It got a bit messy using inlines and I am not quite sure about the result. It feels like to much work for the same result.)
```suggestion
	if (p_func->body->end_line - 2 > p_func->body->line) {
		r_symbol.range.end.line = p_func->body->end_line - 2;
	} else if (p_func->body->line < lines.size()) {
		r_symbol.range.end.line = p_func->body->line;
	} else {
		r_symbol.range.end.line = lines.size() - 1;
	}
```